s3-migrate/test_migrate.py

```diff
@@ -0,0 +1,121 @@
+import tempfile
+from unittest.mock import Mock
+
+import pytest
+from botocore.exceptions import ClientError
+
+from migrate import MogileFile, \
+    md5_fileobj_hex, sha1_fileobj_hex, md5_fileobj_b64, sha1_fileobj_b64
+
+
+# pylint: disable=C0115,C0116,R0201
+class TestMigrate():
+    @pytest.fixture
+    def s3_client(self):
+        s3_client = Mock()
+        obj = Mock()
+        s3_client.Object.return_value = obj
+        obj.load.return_value = {}
+        return s3_client
+
+    @pytest.fixture
+    def mogile_client(self):
+        mogile_client = Mock()
+        paths = Mock()
+        mogile_client.get_paths.return_value = paths
+        paths.data = {
+            'paths': {
+                1: 'http://example.org/1',
+                2: 'http://example.org/2'
+            }
+        }
+        return mogile_client
+
+    @pytest.fixture
+    def row(self):
+        # hello world sha1sum: 2aae6c35c94fcfb415dbe95f408b9ce91ee846ed
+        return {
+            'fid': 1,
+            'dmid': 1,
+            'dkey':
+            '2aae6c35c94fcfb415dbe95f408b9ce91ee846ed-mogilefs-prod-repo',
+            'length': 1593790,
+            'classid': 0,
+            'devcount': 2}
+
+    @pytest.yield_fixture
+    def my_tempfile(self):
+        with tempfile.TemporaryFile() as tmp:
+            tmp.write(b"hello world")
+            yield tmp
```

Comment:
This is cool! I'll have to remember this pattern for using pytest fixtures that represent file objects

Reply:
Yeah, I :heart: pytest